.devcontainer/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM rocker/r-base
+
+RUN apt-get -qq update \
+  && apt-get install -y --no-install-recommends \
+  git libxml2-dev
+
+RUN Rscript -e "
+  read.dcf('DESCRIPTION', 'Imports') |>
```

Comment:
Maybe `remotes::install_deps(dependencies=TRUE)`?

Reply:
It does require installing remotes, which is lightweight but has to be run separately (so `install.packages()` calculates the deps graph twice). Seems fine though.

`dependencies=TRUE` seems too much for dev, I think the default (just `Imports` for us) is better. If you need the others, `install.packages()` works inside the container.

We might also add `Config/needs/testing: patrick` to our DESCRIPTION, then we could combine the 2nd+3rd steps.

Reply:
OK, don't think it'll get simpler than what we have currently. while still using remotes.

A simpler {base} version is this:

```r
read.dcf("DESCRIPTION", "imports") |>
  tools:::.split_dependencies() |>
  names() |>
  c("patrick") |>
  install.packages()
```

That looks pretty good to me and gets us back to one `install.packages()` call, WDYT?

Reply:
Seems a bit convoluted, especially using a `:::` function. Is a separate `install.packages()` call that problematic?
Also, is there a more generic way to specify minimal dev deps? Maybe via some `Config/...` key in DESCRIPTION?

Reply:
FWIW, `.split_dependencies()` hasn't changed substantially in 15+ years:

![image](https://github.com/r-lib/lintr/assets/7606389/da106d94-6167-4be6-a27d-f9dd2f4daf3c)

I really wish they would export it! In fact I filed https://bugs.r-project.org/show_bug.cgi?id=18639.

> Is a separate install.packages() call that problematic?

Not catastrophic, just nice to avoid. In playing around on #2446 yesterday, I noticed the default Travis script does exactly that (install {remotes}, then use it to finish installing deps).

> Also, is there a more generic way to specify minimal dev deps? Maybe via some Config/... key in DESCRIPTION?

Yes, see first comment in this thread (crossed out now). Just wanted to avoid editing DESCRIPTION.

What would you like to change, (1) switch to {remotes} (2) use Config/needs/testing (3) both?

Reply:
Both SGTM if {remotes} takes a flag. This should also lower the entry barrier for new contributors, and we could add a stub our README on how to get a dev environment setup properly.

Reply:
> we could add a stub our README on how to get a dev environment setup properly.

That part, I'm less sure -- IIUC the devcontainers will all run against my billing account 😂 don't necessarily want everyone piling in :)

Reply:
OK, PTAL.

Reply:
> > we could add a stub our README on how to get a dev environment setup properly.
> 
> That part, I'm less sure -- IIUC the devcontainers will all run against my billing account 😂 don't necessarily want everyone piling in :)

I was thinking about how to install the dev deps, not how to use the container 🤣